lib/pure/parsexml.nim

```diff
@@ -446,13 +447,15 @@ proc parseTag(my: var XmlParser) =
     # an attribute follows:
     my.kind = xmlElementOpen
     my.state = stateAttr
-    my.c = my.a # save for later
+    # save for later:
+    my.c.setLen(my.a.len)
+    copyMem(addr my.c[0], addr my.a[0], my.a.len+1)
```

Comment:
Either way, it sucks that with this change it's not evaluable at compile time anymore, so we really should fix the codegen instead.